lib/Doctrine/Common/Annotations/TokenParser.php

```diff
@@ -172,7 +179,7 @@ public function parseUseStatements($namespaceName)
     public function parseNamespace()
     {
         $name = '';
-        while (($token = $this->next()) && ($token[0] === T_STRING || $token[0] === T_NS_SEPARATOR)) {
+        while (($token = $this->next()) && ($token[0] === T_STRING || $token[0] === T_NS_SEPARATOR || (\PHP_VERSION_ID >= 80000 && ($token[0] === T_NAME_QUALIFIED || $token[0] === T_NAME_FULLY_QUALIFIED)))) {
```

Comment:
A linebreak would be welcome here, it will mean less work if we want to enforce use our own CS one day (not sure why we don't yet)

Reply:
what would be the expected formatting ?

Reply:
Several of them are valid I think. I would go with
```suggestion
        while (($token = $this->next()) && ($token[0] === T_STRING || $token[0] === T_NS_SEPARATOR || (
            \PHP_VERSION_ID >= 80000 &&
            ($token[0] === T_NAME_QUALIFIED || $token[0] === T_NAME_FULLY_QUALIFIED)
        ))) {
```

Reply:
done